### PR TITLE
Feature/home blocks  simple post to collection

### PR DESCRIPTION
**Pull request by manuelurenah**
image

image

image

### CODE REVIEW

File: src/client-utils/cf-images-utils.ts [hunk 1]
```diff
@@ -30,5 +30,5 @@ export function getEdgeUrl(src: string, { name, ...variantParams }: Omit<EdgeUrl
   if (name.includes('.')) name = name.split('.').slice(0, -1).join('.') + '.jpeg';
   else name = name + '.jpeg';
 
-  return [env.NEXT_PUBLIC_IMAGE_LOCATION, src, params.toString(), name].join('/');
+  return [env.NEXT_PUBLIC_IMAGE_LOCATION, src, params.toString(), name].filter(Boolean).join('/');
```
Comment (lrojas94) [inline src/client-utils/cf-images-utils.ts:33]: perhaps better use `isDefined`?

File: src/components/Collections/Collection.tsx [hunk 2]
```diff
@@ -111,59 +116,80 @@ export function Collection({
     );
   }
 
+  const collectionType = collection?.type;
+  // TODO.collections: This is tied to images for now but
+  // we will need to add a check for other resources later
+  const canAddContent =
+    collectionType === CollectionType.Image && permissions?.write && permissions?.isContributor;
```
Comment (lrojas94) [inline src/components/Collections/Collection.tsx:123]: need to check for `write` and `writeReview`. No need for `isContributor`, as write/write_review return false when the user is not the owner or a contributor

File: src/server/controllers/collection.controller.ts [hunk 3]
```diff
@@ -233,3 +257,42 @@ export const updateCollectionItemsStatusHandler = async ({
...
+  try {
+    const { user } = ctx;
+    const collection = await getCollectionById({ input: { id: collectionId } });
+    if (!collection) throw throwNotFoundError(`No collection with id ${collectionId}`);
+
+    const permissions = await getUserCollectionPermissionsById({
+      id: collection.id,
+      user,
+    });
+    if (!permissions.write || !permissions.manage)
```
Comment (lrojas94) [inline src/server/controllers/collection.controller.ts:277]: check or `writeReview` here too

File: src/server/services/collection.service.ts [hunk 4]
```diff
@@ -792,3 +807,58 @@ export const updateCollectionItemsStatus = async ({
...
+      collectionId,
+      addedById: user.id,
+    }));
+  }
+  if (postIds.length > 0 && (collection.type === CollectionType.Post || collection.type === null)) {
+    data = postIds.map((postId) => ({
+      postId,
+      collectionId,
+      addedById: user.id,
+    }));
```
Comment (lrojas94) [inline src/server/services/collection.service.ts:860]: if we're considering null types here, we should append to the data, not re-assign. 
Also, be sure to add the relevant status:
```js
status: permission.writeReview
            ? CollectionItemStatus.REVIEW
            : CollectionItemStatus.ACCEPTED,
```

File: src/client-utils/cf-images-utils.ts (hunk 1 above)
Comment (manuelurenah) [reply src/client-utils/cf-images-utils.ts:33]: In this case it's better to use `.filter(Boolean)`, cause we need to check for any falsy value like empty string and strip it out from the components of the url

**Review by lrojas94 — APPROVED** [review]

**Pull request merged**